### PR TITLE
[BUG] TimedeltaIndex intersects with decreasing TimedeltaIndex

### LINKED ISSUE #17391: BUG: TimedeltaIndex.intersection fails for decreasing TimedeltaIndex
#### Code Sample, a copy-pastable example if possible
Setup
```python
In [1]: import pandas as pd
   ...: idx1 = pd.to_timedelta(range(3), unit='s')
   ...: idx2 = pd.to_timedelta(range(2, -1, -1), unit='s')
   ...:

In [2]: idx1
Out[2]: TimedeltaIndex(['00:00:00', '00:00:01', '00:00:02'], dtype='timedelta64[ns]', freq=None)

In [3]: idx2
Out[3]: TimedeltaIndex(['00:00:02', '00:00:01', '00:00:00'], dtype='timedelta64[ns]', freq=None)
```

Intersection returns empty between `idx1` and `idx2`:
```python
In [4]: idx1.intersection(idx2)
Out[4]: TimedeltaIndex([], dtype='timedelta64[ns]', freq=None)

In [5]: idx2.intersection(idx1)
Out[5]: TimedeltaIndex([], dtype='timedelta64[ns]', freq=None)
```

Even an intersection of `idx2` with itself returns empty:
```python
In [6]: idx2.intersection(idx2)
Out[6]: TimedeltaIndex([], dtype='timedelta64[ns]', freq=None)
```

#### Problem description
The intersections are returning empty even though all elements are shared.

#### Expected Output
I'd expect all of the intersections to return a `TimedeltaIndex` containing all three elements.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.2.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 78 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.20.3
pytest: 3.1.2
pip: 9.0.1
setuptools: 27.2.0
Cython: 0.26
numpy: 1.13.1
scipy: 0.19.1
xarray: None
IPython: 6.1.0
sphinx: 1.5.6
patsy: 0.4.1
dateutil: 2.6.0
pytz: 2017.2
blosc: None
bottleneck: 1.2.1
tables: 3.2.2
numexpr: 2.6.2
feather: None
matplotlib: 2.0.2
openpyxl: 2.4.7
xlrd: 1.0.0
xlwt: 1.2.0
xlsxwriter: 0.9.6
lxml: 3.7.3
bs4: 4.6.0
html5lib: 0.999
sqlalchemy: 1.1.9
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by makbigc**
-  closes #17391
-  tests added (two tests taken from #22114)
-  passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
-  whatsnew entry

Beside the bug of decreasing TimedeltaIndex, this PR also solve the following
```
In [2]: idx1 = pd.to_timedelta(range(1,20), unit='s')                           

In [3]: idx2 = pd.to_timedelta((1,2,3,4,5,6,7,8,9,10,20), unit='s')             

In [4]: result = idx1.intersection(idx2)                                        

In [5]: result                                                                  
Out[5]: 
TimedeltaIndex(['00:00:01', '00:00:02', '00:00:03', '00:00:04', '00:00:05',
                '00:00:06', '00:00:07', '00:00:08', '00:00:09', '00:00:10',
                '00:00:11', '00:00:12', '00:00:13', '00:00:14', '00:00:15',
                '00:00:16', '00:00:17', '00:00:18', '00:00:19'],
               dtype='timedelta64[ns]', freq=None)

```
The above is also added into the tests.

### CODE REVIEW

File: doc/source/whatsnew/v0.25.0.rst [hunk 1]
```diff
@@ -141,7 +141,7 @@ Datetimelike
 Timedelta
 ^^^^^^^^^
 
--
+- bug when TimedeltaIndex intersects with another decreasing TimedeltaIndex (:issue:`17391`)
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v0.25.0.rst]: Capitalize bug. Reference the method :meth:\`TimedeltaIndex.intersection\`.

Remove the trailing whitespace.

File: pandas/core/indexes/timedeltas.py [hunk 2]
```diff
@@ -474,6 +474,10 @@ def intersection(self, other):
             return self
         if len(other) == 0:
             return other
+
+        self = self.sort_values()
```
Comment (TomAugspurger) [inline pandas/core/indexes/timedeltas.py]: Sorting can be relatively expensive. Are we able to avoid this?

I think this would also break `sort=False`.

Comment (TomAugspurger) [reply pandas/core/indexes/timedeltas.py]: I would think that we could extract the `i8` values and re-use `Int64Index.sort`. Is that an option?

**Comment by jreback** [comment]
can you merge master

**Pull request closed without merging**